code/modules/mob/living/carbon/inventory.dm

```diff
@@ -155,3 +155,35 @@
 /mob/living/carbon/proc/get_holding_bodypart_of_item(obj/item/I)
 	var/index = get_held_index_of_item(I)
 	return index && hand_bodyparts[index]
+
+/mob/living/carbon/proc/give()
+	var/obj/item/receiving = get_active_held_item()
+	if(!receiving)
+		return
+	visible_message("<span class='notice'>[src] is offering [receiving]</span>", "<span class='notice'You offer [receiving]</span>", null, 2)
+	for(var/mob/living/carbon/C in orange(1, src))
+		var/obj/screen/alert/give/G = C.throw_alert("Give", /obj/screen/alert/give)
+		if(!G)
+			return
+		G.name = "[src] is offering [receiving]"
+		G.desc = "[src] is offering you [receiving]. Click this alert to accept it."
+		G.icon_state = "template"
+		G.cut_overlays()
+		G.add_overlay(receiving)
+		G.receiving = receiving
+		G.giver = src
+		addtimer(CALLBACK(C, .proc/clear_alert, "Give"), 50)

```

Comment:
should probably clear when they move or something, I feel like this can get annoying otherwise

Reply:
(with epic gamer signals)